components/alert/alert-toast.js

```diff
@@ -118,11 +141,45 @@ class AlertToast extends LitElement {
 		}
 	}
 
+	_closeTimerStart() {
+		clearTimeout(this._setTimeoutId);
+		if (!this.noAutoClose && !this._hasFocus && !this._hasMouse) {
+			const duration = this.buttonText ? 10000 : 4000;
+			this._setTimeoutId = setTimeout(() => {
+				this.open = false;
```

Comment:
We need a good place to reset `_hasFocus` and `_hasMouse` to `false`, otherwise if you open and then close and then open it again they'll be their old values. Maybe inside here where we set `open = false` is a good place?

Reply:
If we reset it there it seems like we ought to reset it when we click the close button too, or there'd be a similar problem.

Reply:
Yeah, so maybe on the `updated` when `open` changes eh?

Reply:
Otherwise `elem.open = false` won't reset it.